### PR TITLE
WIP: toward not precomputing the destination edge cost

### LINKED ISSUE #2039: Cost Wraps Arround on Trivial Routes for A*
i'm seeing some cases where the values in the destinations set are creater than the cost up to that point. this leads to a final cost value which is negative. obviously negative cost (while it would be great in real life) is not a good thing. we need to figure out where we are improperly measuring the cost so that when we subtract off the partial cost at the destination we dont underflow the cost.

### PR DESCRIPTION
**Pull request by kevinkreiser**
fixes #2039 

this isnt done yet, i think the computation is still incorrect, but something like this change is what im hoping will get us a gauranteed non-negative result

### CODE REVIEW

File: src/thor/astar.cc [hunk 1]
```diff
@@ -167,7 +167,7 @@ void AStarPathAlgorithm::ExpandForward(GraphReader& graphreader,
     auto p = destinations_.find(edgeid);
     if (p != destinations_.end()) {
       // Subtract partial cost and time
-      newcost -= p->second;
+      newcost *= p->second;
```
Comment (dnesbitt61) [inline src/thor/astar.cc]: this doesn't match the comment, presume you still want -=

Comment (kevinkreiser) [reply src/thor/astar.cc]: ah yeah i didnt change the comment i changed the code to store the percentage along the edge instead of the precomputed cost

**Comment by kevinkreiser** [comment]
closing this @PureW is working on a fix in another branch

**Comment by kevinkreiser** [comment]
replaced by: https://github.com/valhalla/valhalla/pull/2120

**Pull request closed without merging**